### PR TITLE
Unit Tests For Caldera_Forms::run_calculation()

### PR DESCRIPTION
**Pull request by New0**
This will replace deprecated create_function() by a math parser.

This is work for #2335

### CODE REVIEW

File: tests/test-run-calculation.php [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+            'fld_93143' => '1:1',
+            'fld_7465068' => '1:1',
+            'fld_8568604' => '1:1',
+            'fld_3665187' => '1:1',
+          ),
+          'structure' => '12',
+        ),
+        'fields' => 
+        array(
+          'fld_1892141' =>
```
Comment (Shelob9) [inline tests/test-run-calculation.php]: @New0 You have some of the fields in the form config using fld_... for ID and some using a meaningful ID. Let's make them all meaningful and mayb use class properties. `$this->IdOdFieldThatHasValueOf1;  $this->manualCalcFieldId`

File: tests/test-run-calculation.php [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+        $field = $form['fields']['fld_8568604'];
+
+        Caldera_Forms::set_field_data('fld_1892141', $fieldOneValue, $form );
+        Caldera_Forms::set_field_data('fld_93143', $fieldTwoValue, $form );
+        Caldera_Forms::set_field_data('fld_7465068', $fieldTwoValue, $form );
+        
+        $result = ($fieldOneValue + $fieldTwoValue) / $fieldThreeValue;
+
+        //Test Calculation by fields
+        $cf_calc = Caldera_Forms::run_calculation( null, $field, $form );
```
Comment (Shelob9) [inline tests/test-run-calculation.php]: @New0 Small point, but changed `$cf_calc` to `$expectedResult` that would be more obvious when reviewing the test later.

**Comment by New0** [comment]
Thanks for your comments @Shelob9 , it helped me realize a typo was making the test to fail

**Comment by New0** [comment]
( @Shelob9 run composer install to test this )

File: classes/core.php [hunk 3]
```diff
@@ -1446,8 +1450,17 @@ static public function run_calculation($value, $field, $form)
...
-		$total = $total_function();
+		/**
+		 *  Set result out of formula Parser 
+		 *
+		 * @since 1.8.5
+		 * 
+		 * @param $formula 
+		 * @param $precision non required, Number of digits after the decimal point
+		 */
+		$parser = new FormulaParser($formula);
```
Comment (Shelob9) [inline classes/core.php]: @New0 Let's revert this until we have all of the tests in place. Once they pass as-is, then we swap out the parser.

**Comment by Shelob9** [comment]
As we discussed yesterday, this PR will NOT fix #2335 But the tests that @New0 wrote are important  valuable, no matter how we solve #2335 and also useful to have. So this PR is now for #3203

**Comment by Shelob9** [comment]
@New0 I think that this PR is blocked by:

- [ ] Need to reset to c837f5c
- [ ] Remove parser as dependency

Right? If so, please do and then request review again.

**Comment by New0** [comment]
@Shelob9 correct for the Math parser, it was still listed in composer requirements. For the eval() function, it was already reverted and the core.php file is back to deprecated create_function() function

**Pull request merged**